rust_snuba/rust_arroyo/src/processing/mod.rs

```diff
@@ -67,9 +68,11 @@ impl<TPayload: 'static + Clone> AssignmentCallbacks for Callbacks<TPayload> {
             None => {}
             Some(s) => {
                 s.close();
-                // TODO: We need to actually call consumer.commit() with the commit request.
-                // Right now we are never committing during consumer shutdown.
-                let _ = s.join(None);
+                if let Ok(Some(commit_request)) = s.join(None) {
+                    let mut consumer = self.consumer.lock().unwrap();
+                    consumer.stage_offsets(commit_request.positions).unwrap();
+                    consumer.commit_offsets().unwrap();
```

Comment:
I think you should really start raising errors appropriately instead of panicking sooner than later. It will be really hard to hunt down all these places later, the more logic is added. The likely result is that some consumers will panic in prod.

Reply:
Will do it soon

Reply:
it is semantically more correct to propagate as `Result::Err`, but even in python we have no concrete idea what to do with those errors except crash the process.